socialstream/utils.py

```diff
@@ -0,0 +1,127 @@
+from typing import TypedDict
+
+from sotopia.database import EpisodeLog
+
+
+class messageForRendering(TypedDict):
+    role: str
+    type: str
+    content: str
+
+
+def parse_reasoning(reasoning: str, num_agents: int) -> tuple[list[str], str]:
+    """Parse the reasoning string into a dictionary."""
+    sep_token = "SEPSEP"
+    for i in range(1, num_agents + 1):
+        reasoning = (
+            reasoning.replace(f"Agent {i} comments:\n", sep_token)
+            .strip(" ")
+            .strip("\n")
+        )
+    all_chunks = reasoning.split(sep_token)
+    general_comment = all_chunks[0].strip(" ").strip("\n")
+    comment_chunks = all_chunks[-num_agents:]
+
+    return comment_chunks, general_comment
+
+
+def render_for_humans(episode: EpisodeLog) -> list[messageForRendering]:
```

Comment:
It's best if we leave this function to sotopia instead of here.

Reply:
let's leave it there for now

Reply:
but we do have a separate `render_for_humans` in sotopia?

Reply:
hmmm, let's leave it there for now.